cmd_run.go

```diff
@@ -276,8 +285,49 @@ func RunAction(c *cli.Context) error {
 	return nil
 }
 
+// GetIpfsVersion tries to retrieve the IPFS version using the IPFS API client.
+// If it fails, it falls back to an HTTP request to the specified endpoint, using host and port from the context.
+func (t *tiros) GetIpfsVersion(c *cli.Context) (version string, sha string, err error) {
+	version, sha, err = t.ipfs.Version()
+	if err == nil {
+		return version, sha, nil
+	}
+
+	// Use host and port from the context for the fallback HTTP request
+	apiHost := c.String("ipfs-api-host")
+	apiPort := c.Int("ipfs-api-port")
+	apiUrl := fmt.Sprintf("http://%s:%d/api/v0/version", apiHost, apiPort)
+	// log that we're falling back to HTTP url for version
+	log.WithField("url", apiUrl).Debugln("Falling back to HTTP request for version")
+
+	resp, httpErr := http.Get(apiUrl)
+	if httpErr != nil {
+		return "", "", fmt.Errorf("ipfs api and http fallback both failed: %v, fallback error: %w", err, httpErr)
+	}
+	defer resp.Body.Close()
+
+	body, readErr := io.ReadAll(resp.Body)
+	if readErr != nil {
+		return "", "", fmt.Errorf("error reading version endpoint response: %w", readErr)
+	}
+	// log the entire body of the response
+	log.WithField("body", string(body)).Debugln("HTTP response body")
+
+	var data struct {
+		Version string `json:"Version"`
+		Commit  string `json:"Commit"`
+	}
+
+	jsonErr := json.Unmarshal(body, &data)
+	if jsonErr != nil {
+		return "", "", fmt.Errorf("error parsing version endpoint JSON: %w", jsonErr)
+	}
+
+	return data.Version, data.Commit, nil
```

Comment:
Under the hood `t.ipfs.Version()` is doing almost the same HTTP request to `fmt.Sprintf("http://%s:%d/api/v0/version", apiHost, apiPort)`. The client is not using gRPC or anything. The only difference here seems to be that the Kubo HTTP endpoints expect a `POST` request [[docs](https://docs.ipfs.tech/reference/kubo/rpc/#api-v0-version)]. Therefore, `t.ipfs.Version()` is also doing a `POST` instead of a `GET` as in line 303.

Would it work if we changed helia to also expect a `POST` instead of a `GET` request?

If that _is_ possible AND we really need the `ipfs-api-host` flag from above, I think we could get away with just changing:

```go
ipfsClient = shell.NewShell(fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", c.Int("ipfs-api-port")))
```

to

```
ipfsClient = shell.NewShell(fmt.Sprintf("/ip4/%s/tcp/%d", c.String("ipfs-api-host"), c.Int("ipfs-api-port")))
```

Reply:
Post requests for version are also supported in helia-http-gateway and this fails against helia-http-gateway when ran locally. Maybe due to docker networking on mac?